client/me/billing-history/receipt.jsx

```diff
@@ -75,69 +79,72 @@ module.exports = React.createClass( {
 						onClick={ this.recordClickEvent( 'Print Receipt Button in Billing History Receipt', this.printReceipt ) }
 						className="button is-secondary"
 					>
-						{ this.translate( 'Print Receipt' ) }
+						{ translate( 'Print Receipt' ) }
 					</a>
 				</Card>
 			</Main>
 		);
 	},
 
-	printReceipt: function( event ) {
+	printReceipt( event ) {
 		event.preventDefault();
 		window.print();
 	},
 
-	ref: function() {
-		if ( ! this.props.transaction.pay_ref ) {
+	ref() {
+		const { transaction, translate } = this.props;
+
+		if ( ! transaction.pay_ref ) {
 			return null;
 		}
 
 		return (
 			<li>
-				<strong>{ this.translate( 'Transaction ID' ) }</strong>
-				<span>{ this.props.transaction.pay_ref }</span>
+				<strong>{ translate( 'Transaction ID' ) }</strong>
+				<span>{ transaction.pay_ref }</span>
 			</li>
 		);
 	},
 
-	paymentMethod: function() {
-		var transaction = this.props.transaction,
-			text;
+	paymentMethod() {
+		const { transaction, translate } = this.props;
+		let text;
 
 		if ( transaction.pay_part === 'paypal_express' ) {
-			text = this.translate( 'PayPal' );
+			text = translate( 'PayPal' );
 		} else if ( 'NOT STORED' === transaction.cc_type.toUpperCase() ) {
-			text = this.translate( 'Credit Card' );
+			text = translate( 'Credit Card' );
 		} else {
-			text = transaction.cc_type.toUpperCase() + this.translate( ' ending in ' ) + transaction.cc_num;
+			text = transaction.cc_type.toUpperCase() + translate( ' ending in ' ) + transaction.cc_num;
 		}
 
 		return (
 			<li>
-				<strong>{ this.translate( 'Payment Method' ) }</strong>
+				<strong>{ translate( 'Payment Method' ) }</strong>
 				<span>{ text }</span>
 			</li>
 		);
 	},
 
-	renderBillingDetails: function( transaction ) {
-		if ( ! this.props.transaction.cc_name && ! this.props.transaction.cc_email ) {
+	renderBillingDetails() {
+		const { transaction, translate } = this.props;
+		if ( ! transaction.cc_name && ! transaction.cc_email ) {
 			return null;
 		}
 
 		return (
 			<li className="billing-history__billing-details">
-				<strong>{ this.translate( 'Billing Details' ) }</strong>
+				<strong>{ translate( 'Billing Details' ) }</strong>
 				<div contentEditable="true">{ transaction.cc_name }</div>
```

Comment:
@gwwar these two fields are displayed only when that purchase has been made with a credit card. And they allow the user to change the name or email directly in the receipt before printing it. Here is a preview of how the editable fields look like when rolled over with the mouse: 

![](https://cldup.com/qRUej5b9se.png)

This appears to have been introduced in 6648-gh-calypso-pre-oss.